pallets/random-seed/src/lib.rs

```diff
@@ -0,0 +1,122 @@
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::{
+    decl_module, decl_storage,
+    weights::DispatchClass,
+};
+use sp_inherents::{InherentData, InherentIdentifier, ProvideInherent, IsFatalError};
+use sp_runtime::RuntimeString;
+use sp_std::result;
+use sp_core::H256;
+use codec::Encode;
+use codec::Decode;
+
+#[cfg(feature = "std")]
+use sp_inherents::ProvideInherentData;
+
+/// The module configuration trait
+pub trait Trait: frame_system::Trait {
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+
+        #[weight =  (
+            10_000,
+            DispatchClass::Mandatory
+        )]
+        fn set(origin, seed: H256) {
+            <Self as Store>::Seed::put(seed);
+        }
+
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as Timestamp {
+        /// Current time for the current block.
+        pub Seed get(fn seed) : H256;
+    }
+}
+
+impl<T: Trait> Module<T> {
+    pub fn get() -> H256 {
+        Self::seed()
+    }
+}
+
+// originally in sp-module
+pub type RandomSeedInherentType = H256;
+const RANDOM_SEED_INHERENT_IDENTIFIER: InherentIdentifier = *b"blckseed";
+
+#[derive(Encode, sp_runtime::RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(Decode))]
+pub enum RandomSeedInherentError {
```

Comment:
why we need this error enum and following error implementations?

Reply:
this is how inherents works in substrate. You implement `ProvideInherent` trait for given pallet and expose Inherent features in runtime. Block builder operates on inherent data passed from babe, those data can be easily registered anywhere as those are just mappings 
`identifier -> data`  then later on in runtime pallets that implements ProvideInherent tries to fetch those 
`identifier -> data` mappings, deserialize stored information(`H256` seed in our case) and automatically triggers proper extrinsic ( storage `Seed::set` ) to store seed in blockchain. But such deserialization may fail thats why proper errors needs to be defined and triggered